cirq-core/cirq/sim/simulator.py

```diff
@@ -659,6 +648,7 @@ def simulate_moment_steps(
         qubits = ops.QubitOrder.as_qubit_order(qubit_order).order_for(circuit.all_qubits())
         return self._base_iterator(resolved_circuit, qubits, actual_initial_state)
 
+    @abc.abstractmethod
```

Comment:
Is it possible that subclasses don't have this function defined?

Reply:
No, if they didn't have an implementation then they'd have received the warning from the default implementation on line 683.

> Custom implementations of `cirq.SimulatesIntermediateState` should implement
             `_base_iterator` directly rather than implementing both `_create_simulation_state`
             and `_core_iterator`.